cassandra-five-zero-bridge/src/test/java/org/apache/cassandra/spark/reader/SSTableCacheTests.java

```diff
@@ -191,4 +201,35 @@ public void testCache()
                 }
             });
     }
+
+    @Test
+    void testNoLeakDetectedError() throws Exception
+    {
+        Partitioner partitioner = Partitioner.Murmur3Partitioner;
+        try (TemporaryDirectory directory = new TemporaryDirectory())
+        {
+            // Write an SSTable
+            TestSchema schema = TestSchema.basic(BRIDGE);
+            schema.writeSSTable(directory, BRIDGE, partitioner,
+                                writer -> IntStream.range(0, 10).forEach(index -> writer.write(index, 0, index)));
+            SSTable sstable = TestSSTable.firstIn(directory.path());
+            TableMetadata metadata = new SchemaBuilder(schema.createStatement,
+                                                       schema.keyspace,
+                                                       new ReplicationFactor(ReplicationFactor.ReplicationStrategy.SimpleStrategy,
+                                                                             ImmutableMap.of("replication_factor", 1)),
+                                                       partitioner).tableMetaData();
+
+            SummaryDbUtils.Summary summary = SSTableCache.INSTANCE.keysFromSummary(metadata, sstable);
+            assertThat(summary).isNotNull();
+            assertThat(SSTableCache.INSTANCE.containsSummary(sstable)).isTrue();
+            SSTableCache.INSTANCE.invalidate(sstable);
+            summary = null;
+            assertThat(SSTableCache.INSTANCE.containsSummary(sstable)).isFalse();
+            // trigger GC and wait a bit before asserting LEAK DETECTED is not logged.
+            System.gc();
+            Thread.sleep(1000);
```

Comment:
NIT: We could use [Awaitility](https://mvnrepository.com/artifact/org.awaitility/awaitility), if we consider more places that need waiting for asynchronous condition. Kafka developers do not pull the dependency (even though only for test), but implemented a simple function to do the same [here](https://github.com/apache/kafka/blob/trunk/core/src/test/scala/unit/kafka/utils/TestUtils.scala#L712).